### DISCUSSION TITLE
Mouse not working in PyCharm 2023.1 on Windows 10

### DISCUSSION THREAD

**Discussion by ghost** [Textualize/textual#2531] [Q&A] [answered]
I've tried running the textual demo in PyCharm.
Keyboard Navigation etc. works fine.
However, the mouse doesn't work at all.
If there is supposed to be a cursor it isn't there.
If clicks are supposed to work, they don't.
I tried cmd.exe, PowerShell and git bash type of Terminal Windows inside PyCharm.

Is this to be expected?

```
textual diagnose

```

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.24.1 |
| Rich    | 13.3.5 |

## Python

| Name           | Value                                                                                            |
|----------------|--------------------------------------------------------------------------------------------------|
| Version        | 3.9.13                                                                                           |
| Implementation | CPython                                                                                          |
| Compiler       | MSC v.1929 64 bit (AMD64)                                                                        |
| Executable     | -censored-|

## Operating System

| Name    | Value      |
|---------|------------|
| System  | Windows    |
| Release | 10         |
| Version | 10.0.19045 |

## Terminal

| Name                 | Value     |
|----------------------|-----------|
| Terminal Application | *Unknown* |
| TERM                 | *Not set* |
| COLORTERM            | *Not set* |
| FORCE_COLOR          | *Not set* |
| NO_COLOR             | *Not set* |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=332, height=15 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 332                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 15                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

**Comment by davep** [comment]
Unfortunately this is a shortcoming of the PyCharm terminal; there are a few issues with it I'm afraid.

**Comment by ghost** [comment]
thats's unfortunate, I'm mostly working with PyCharm Terminals...
Means I can't use Textualize

**Reply by davep** [reply, marked as answer]
For what it's worth, I tested this on macOS and Windows 11 (the two environments I have that I can install PyCharm into) and the mouse works fine in the terminal, although it's worth noting that the PyCharm terminal on Windows 11 isn't a great experience due to how it handles fonts:

<img width="1208" alt="Screenshot 2023-05-10 at 07 28 23" src="https://github.com/Textualize/textual/assets/28237/2d102a95-020c-4719-8b1d-d1fbcb39e617">

The PyCharm terminal on macOS seems to handle fonts just fine:

![Screenshot 2023-05-10 at 07 33 39](https://github.com/Textualize/textual/assets/28237/0183a28e-828e-4fb6-bf09-3dd72620fd08)

I seem to recall that there's no mouse support in the VSCode terminal on Windows 10 either, but it's fine on Windows 11. So on the mouse front my suspicion here would be that terminal mouse support is lacking.
